lib/ansible/modules/cloud/amazon/elasticsearch.py

```diff
@@ -0,0 +1,377 @@
+
+    if existing_access_json != supplied_access_json:
+        modifications_needed['AccessPolicies'] = module.params['access_policies']
+
+    return modifications_needed
+
+
+def main():
+    argument_spec = ec2.ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            domain={'required': True},
+            version={'required': False, 'default': '5.1'},
+            instance_count={'required': False, 'type': 'int', 'default': 1},
+            instance_type={
+                'required': False, 'default': 't2.medium.elasticsearch'
+            },
+            zone_awareness={
+                'required': False, 'type': 'bool', 'default': True
+            },
+            dedicated_master_enabled={
+                'required': False, 'type': 'bool', 'default': False,
+                'choices': [True, False]
+            },
+            dedicated_master_type={'required': False},
+            dedicated_master_count={'required': False, 'type': 'int'},
+            ebs_volume_type={
+                'required': False, 'default': 'gp2',
+                'choices': ['standard', 'gp2', 'io1']
+            },
+            ebs_volume_size={'required': True, 'type': 'int'},
+            access_policies={'required': True},
+            wait={'required': False, 'type': 'bool'},
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    if not ec2.HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    region, ec2_url, aws_connect_kwargs = ec2.get_aws_connection_info(module)
+
+    if region:
+        try:
+            client = ec2.boto3_conn(
+                module, 'client', 'es', region,
+                profile_name=aws_connect_kwargs['profile_name'])
+        except Exception as e:
+            module.fail_json(msg=str(e))
+    else:
+        module.fail_json(msg='region must be specified')
+
+    changed, domain = ensure_present(client, module)
+    modifications_needed = check_existing_config(module, domain)
+
+    if modifications_needed:
+        changed = True
+        modifications_needed['DomainName'] = module.params['domain']
+        client.update_elasticsearch_domain_config(**modifications_needed)
+
+        # Reload after modifying
+        domain = is_present(client, module.params['domain'])
+
+    if module.params['wait']:
+        while not domain['DomainStatus'].get('Endpoint'):
+            time.sleep(10)
+            domain = is_present(client, module.params['domain'])
+
+    domain_status = domain['DomainStatus']
+    cluster_config_status = domain_status['ElasticsearchClusterConfig']
+    ebs_options = domain_status['EBSOptions']
+
+    module.exit_json(
+        changed=changed,
+        arn=domain_status['ARN'],
+        endpoint=domain_status.get('Endpoint'),
+        processing=domain_status['Processing'],
+        elasticsearch_version=domain_status['ElasticsearchVersion'],
```

Comment:
I'm not sure, since it looks like you're picking info from two result dicts, but the camel_dict_to_snake_dict function might help you out for these lines - you could call that for both dicts and them combine them.